components/ipfs/keys/ipns_keys_manager.cc

```diff
@@ -49,12 +76,12 @@ void IpnsKeysManager::RemoveKey(const std::string& name,
   GURL gurl =
       net::AppendQueryParameter(generate_endpoint, kArgQueryParam, name);
 
-  auto url_loader = CreateURLLoader(gurl, "POST");
+  auto url_loader = ipfs::CreateURLLoader(gurl, "POST");
 
   auto iter = url_loaders_.insert(url_loaders_.begin(), std::move(url_loader));
   iter->get()->DownloadToStringOfUnboundedSizeUntilCrashAndDie(
       url_loader_factory_.get(),
-      base::BindOnce(&IpnsKeysManager::OnKeyRemoved, base::Unretained(this),
+      base::BindOnce(&IpnsKeysManager::OnKeyRemoved, weak_factory_.GetWeakPtr(),

```

Comment:
Here `weak_factory_` is redundant, because the manager owns url loaders, and it will destroy loaders in its destructor. Loaders cancel their callbacks when destroyed